# Novel pipeline to detect Spliced XBP1 from Bulk RNA-Sequencing Data

**Background**

Endoplasmic reticulum (ER) Stress, caused by accumulation of unfolded proteins in the ER activates a key homeostatic pathway called the‚ÄØUnfolded Protein Response (UPR)‚ÄØto stabilize‚ÄØthe‚ÄØcell‚Äôs physiological activities. However prolonged effects of ER stress and deregulation of the UPR as a result, has been shown to play an important role in tumorgenesis and tumor progression in some cancers1,2.‚ÄØOut of the three known UPR activation mechanisms, the most conserved‚ÄØis the inositol-requiring enzyme 1A (IRE1ùõº) and X-box binding protein 1 (XBP1) interaction pathway.‚ÄØThis pathway is activated upon encountering‚ÄØER stress,‚ÄØwhich leads to induction of a very specific‚ÄØ26-nucleotide‚ÄØdeletion in‚ÄØthe‚ÄØcanonical XBP1‚ÄØgene by phosphorylated IRE1ùõº‚ÄØto create the spliced XBP1(XBP1s). XBP1s is the active transcription factor (TF) which is known to regulate key cell fate pathways downstream and can serve as an important therapeutic target3. However, given the frameshift is relatively infinitesimal it is challenging to simply quantify it as an isoform. As a result, the current and only best method to measure the active XBP1s TF relies on qPCR-based quantification and some other in vitro methods4. While this technique has proven to successfully quantify XBP1s levels in smaller sample sets, it would be an extensive effort to quantify the XBP1s levels in larger datasets, such as patient cohorts or functional genomic screens to study effects of ER stress activation. Therefore, building upon our knowledge of ER stress‚ÄØmechanism and quantification techniques, we curated a computational pipeline to detect XBP1s‚ÄØlevels‚ÄØdirectly from RNA-Seq data,‚ÄØas an alternate method.‚ÄØThis method of computationally quantifying spliced percentage‚ÄØlevels of XBP1s can be‚ÄØused as scores to make associations with‚ÄØrelevant ER stress‚ÄØspecific molecular and clinical outcomes.‚ÄØ We further apply this pipeline to explore the XBP1s patterns across the pan cancer TCGA dataset to identify subsets of cancer types and key cancer-type specific molecular signatures. Overall, we propose the utility of the XBP1s detection pipeline as a resource for future applications to various RNA-Seq data not just for cancer datasets but across multiple diseases for study of ER Stress activation patterns.  


**Methods**
- A combination of open-source, command line-based bioinformatics tools to process and quantify the indel from raw RNASeq FASTQ
- customized reference for the alignment was generated using the XBP1 gene sequence for Human assemble hg19 and Mus Musculus mm10 RefSeq databases. As gene expression reference we used XBP1 gene with the 1820 bp in length human ortholog (RefSeq id: NM_005080) and 2264 bp Mouse ortholog (RefSeq id: NM_103842.3)
- Using the Karkkainen‚Äôs‚ÄØblockwise algorithm based bowtie2-build, the index files for the gene sequence were generated respectively
- The alignment was then performed using bowtie2 aligner, with parameters optimized to make it more sensitive
- instead of local alignment, end-to-end alignment was specified for a relatively lenient alignment for sensitiving pipeline to detecting indels
- Each of the parameters were tested with combinations configured in increments on validation RNASeq data, prior to establishing the selected ones. Below are values for all specified parameters. 

*bowtie2 -p 8 -D 20 -R 3 -N 0 -L 20 -i S,1,0.50 --end-to-end --rdg 1,1 --rfg 1,1 --gbar 10* 
